rust/vectordb/src/query.rs

```diff
@@ -69,11 +69,13 @@ impl Query {
     pub async fn execute(&self) -> Result<DatasetRecordBatchStream> {
         let mut scanner: Scanner = self.dataset.scan();
 
-        scanner.nearest(
-            crate::table::VECTOR_COLUMN_NAME,
-            &self.query_vector,
-            self.limit,
-        )?;
+        if let Some(query) = self.query_vector.as_ref() {
+            // If there is a vector query, default to limit=10 if unspecified
+            scanner.nearest(crate::table::VECTOR_COLUMN_NAME, query, self.limit.unwrap_or(10))?;
+        } else if let Some(limit) = self.limit {
+            // If there is no vector query, it's ok to not have a limit
+            scanner.limit(Some(limit as i64), None)?;
+        }
```

Comment:
Minor nit:  I think this can maybe be simplified to...

```
} else {
  scanner.limit(self.limit,None)?;
}
```

but I'm not entirely certain it's the same thing.

Reply:
i changed it to `scanner.limit(self.limit.map(|limit| limit as i64, None)?`